packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,215 @@
+   * @extends {Blockly.FieldNumber}
+   * @constructor
+   */
+  constructor(opt_value, opt_min, opt_max, opt_precision,
+      opt_validator, opt_config) {
+    super(opt_value, opt_min, opt_max, opt_precision, opt_validator, opt_config);
+
+    /**
+     * Array holding info needed to unbind events.
+     * Used for disposing.
+     * Ex: [[node, name, func], [node, name, func]].
+     * @type {!Array.<Array<?>>}
+     * @private
+     */
+    this.boundEvents_ = [];
+
+    /**
+     * The HTML range input element.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Creates and shows the slider field's editor.
+   * @protected
+   * @override
+   */
+  showEditor_() {
+    super.showEditor_();
+    // Build the DOM.
+    var editor = this.dropdownCreate_();
+
+    Blockly.DropDownDiv.getContentDiv().appendChild(editor);
+
+    Blockly.DropDownDiv.setColour(this.sourceBlock_.style.colourPrimary,
+        this.sourceBlock_.style.colourTertiary);
+
+    Blockly.DropDownDiv.showPositionedByField(
+        this, this.dropdownDispose_.bind(this));
+  }
+
+  /**
+   * Creates the slider editor and add event listeners.
+   * @return {!Element} The newly created slider.
+   * @private
+   */
+  dropdownCreate_() {
+    var wrapper = document.createElement('div');
+    var sliderInput = document.createElement('input');
+    sliderInput.setAttribute('type', 'range');
+    sliderInput.setAttribute('min', this.min_);
+    sliderInput.setAttribute('max', this.max_);
+    sliderInput.setAttribute('value', this.getValue());
+    sliderInput.className ='fieldSlider';
+    wrapper.appendChild(sliderInput);
+    this.sliderInput_ = sliderInput;
+
+    this.boundEvents_.push(Blockly.bindEventWithChecks_(
+        sliderInput, 'input', this, this.onSliderChange_));
+
+    return wrapper;
+  }
+
+  /**
+   * Disposes of events belonging to the slider editor.
+   * @private
+   */
+  dropdownDispose_() {
+    for (let event of this.boundEvents_) {
+      Blockly.unbindEvent_(event);
+    }
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Updates the slider when the field rerenders.
+   * @protected
+   * @override
+   */
+  render_() {
+    super.render_();
+    this.updateSlider_();
+  }
+
+  /**
+   * Sets the text to match the slider's position.
+   * @private
+   */
+  onSliderChange_() {
+    this.setEditorValue_(this.sliderInput_.value);
+  }
+
+  /**
+   * Updates the slider when the field rerenders.
+   * @private
+   */
+  updateSlider_() {
+    if (!this.sliderInput_) {
+      return;
+    }
+    this.sliderInput_.setAttribute('value', this.getValue());
+  };
+
+}
+
+/**
+ * Constructs a FieldSlider from a JSON arg object.
+ * @param {!Object} options A JSON object with options (value, min, max, and
+ *                          precision).
+ * @return {!FieldSlider} The new field instance.
+ * @package
+ * @nocollapse
+ */
+FieldSlider.fromJson = function(options) {
```

Comment:
You can move this to the class. Just make it a static method, eg:
```js
static fromJson(options) {
    return new FieldSlider(options['value'],
        undefined, undefined, undefined, undefined, options);
  };
```

Reply:
Done